### PR TITLE
Feat: skip croniter when possible for performance

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlmesh/utils/cron.py [hunk 1]
```diff
@@ -21,12 +23,37 @@ def cron_prev(cron: str, time: TimeLike) -> float:
...
+        self.curr: TimeLike = now() if time is None else time
+        self._interval_seconds: t.Optional[int] = None
+
+    @property
+    def interval_seconds(self) -> int:
+        if self._interval_seconds is None:
+            seconds = set()
+            curr = to_datetime(self.curr)
+
+            for _ in range(10):
```
Comment (izeigerman) [inline sqlmesh/utils/cron.py:35]: [Nit] Maybe a named constant like `ESTIMATE_SAMPLES_NUM` or something like that.

File: sqlmesh/utils/cron.py [hunk 2]
```diff
@@ -21,12 +23,37 @@ def cron_prev(cron: str, time: TimeLike) -> float:
...
+                prev = curr
+                curr = to_datetime(cron_next(self.cron, curr))
+                seconds.add(curr - prev)
+
+            if len(seconds) == 1:
+                self._interval_seconds = first(seconds).seconds
+            else:
+                self._interval_seconds = 0
```
Comment (izeigerman) [inline sqlmesh/utils/cron.py:43]: Why 0?

Comment (izeigerman) [reply sqlmesh/utils/cron.py:43]: Because it's not `None` but still behaves as false? I'd prefer this to be more explicit

Comment (tobymao) [reply sqlmesh/utils/cron.py:52]: docstring

**Review by izeigerman — APPROVED** [review]

**Pull request merged**